src/main/java/hudson/plugins/promoted_builds/JobPropertyImpl.java

```diff
@@ -79,6 +79,13 @@ public JobPropertyImpl(AbstractProject<?,?> owner) throws Descriptor.FormExcepti
         init();
     }
 
+    /**
+     * Programmatic construction.
+     */
+    public JobPropertyImpl(Set<String> activeProcessNames) throws Descriptor.FormException, IOException {
+        this.activeProcessNames.addAll(activeProcessNames);
```

Comment:
I would move validation/modification code from another constructor here.
Otherwise you can remove exception throwing and annotate this constructor as <code>@Restricted(NoExternalUse.class)</code>